masonry/src/tests/event.rs

```diff
@@ -153,4 +153,100 @@ fn click_anchors_focus() {
 
 // TEXT EVENTS
 
+#[test]
+fn text_event() {
+    let target_tag = WidgetTag::new("target");
+
+    let target = NewWidget::new_with_tag(TextArea::new_editable("").record(), target_tag);
+
+    let mut harness = TestHarness::create(default_property_set(), target);
+    let target_id = harness.get_widget_with_tag(target_tag).id();
+    harness.flush_records_of(target_tag);
+    harness.keyboard_type_chars("A");
+    harness.flush_records_of(target_tag);
+
+    // The widget isn't focused, it doesn't get text events.
+    harness.keyboard_type_chars("A");
+    assert_matches!(harness.get_records_of(target_tag)[..], []);
+
+    // We focus on the widget, now it gets text events.
+    harness.focus_on(Some(target_id));
+    harness.keyboard_type_chars("A");
+    let records = harness.get_records_of(target_tag);
+    assert!(records.iter().any(|r| matches!(r, Record::TextEvent(_))));
+}
+
+#[test]
+fn text_event_fallback() {
```

Comment:
It might be worth noting that this is interim behaviour/that a better fallback mechanism is desired.

Is there an issue for that?

Reply:
Well there's the #1269 PR.